### PR TITLE
Major refactor on poseidon tree for microkelvin

### PR DESCRIPTION
**Pull request by vlopes11**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
+version = "0.11.0"
 authors = [
   "zer0 <matteo@dusk.network>",	"vlopes11 <victor@dusk.network>",	"CPerezz <carlos@dusk.network>", "Kristoffer Ström <kristoffer@dusk.network>"
 ]
 edition = "2018"
 
 [dependencies]
-kelvin = "0.19.0"
-nstack = "0.5"
+nstack = { git = "https://github.com/dusk-network/nstack", branch = "vlopes11/bump-microkelvin" }
```
Comment (CPerezz) [inline Cargo.toml]: This won't be merged until the git dep becomes a tag. I guess is pending the merge of: https://github.com/dusk-network/nstack/pull/15

File: src/tree/annotation.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
+use canonical::{Canon, Store};
+use canonical_derive::Canon;
+use core::borrow::Borrow;
+use dusk_plonk::prelude::BlsScalar;
```
Comment (CPerezz) [inline src/tree/annotation.rs]: It's just a minor thing, but should't prelude be always fully imported? Otherways, you should get the scalar from the dusk_plonk::bls12_381::Scalar as BlsScalar.

File: src/tree/branch.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            .rev()
+            .zip(branch.0.iter_mut())
+            .for_each(|(l, b)| {
+                depth += 1;
+                b.offset = l.offset() + 1;
+
+                let mut flag = 1;
+                let mut mask = 0;
+
+                match &**l {
```
Comment (CPerezz) [inline src/tree/branch.rs:101]: This looks a bit odd. Wouldn't it be more readable if we call as_ref or deref here?

**Review by CPerezz — APPROVED** [review]
This looks really good to me! Good job @vlopes11 :smile:

File: Cargo.toml (hunk 1 above)
Comment (vlopes11) [reply Cargo.toml]: Fixed

File: src/tree/annotation.rs (hunk 2 above)
Comment (vlopes11) [reply src/tree/annotation.rs]: Fixed

**Review by krl — APPROVED** [review]
looks good!

File: src/tree/branch.rs (hunk 3 above)
Comment (vlopes11) [reply src/tree/branch.rs:101]: I agree this style is weird and ugly, but as discussed we don't really have a better option for rust

**Comment by CPerezz** [comment]
@vlopes11 feel free to merge!

**Pull request merged**